GoogleUtilities/Tests/Unit/Environment/GULHeartbeatDateStorageTest.m

```diff
@@ -198,10 +296,101 @@ - (void)testSetHeartbeatDateForTagWhenHeartbeatFileReturnsInvalidData {
+- (void)testSetHeartbeatDateForTagWhenStoredHeartbeatDictionaryIsImmutable {
+  // 1. Manually create the heartbeat directory.
+  NSURL *heartbeatStorageDirectoryURL = [self pathURLForDirectory:kGULHeartbeatStorageDirectory];
+  NSError *directoryError;
+  BOOL directoryCreated = [self explicitlyCreateDirectoryForURL:heartbeatStorageDirectoryURL
+                                                      withError:&directoryError];
+  XCTAssert(directoryCreated);
+
+  // 2. Populate the storage file with heartbeat info.
+  // 2.1 Create an *immutable* dictionary with heartbeat info.
+  NSDate *storedDate = [NSDate distantPast];
+  NSString *storedTag = @"stored-tag";
+  NSDictionary *storedHeartbeatDictionary = @{storedTag : storedDate};
+  // 2.2 Encode the dictionary.
+  NSError *archiveError;
+  NSData *data = [GULSecureCoding archivedDataWithRootObject:storedHeartbeatDictionary
+                                                       error:&archiveError];
+  XCTAssertNotNil(data);
+  XCTAssertNil(archiveError);
+  // 2.3 Write the encoded dictionary to file.
+  NSURL *heartbeatStorageFileURL = [self fileURLForDirectory:heartbeatStorageDirectoryURL];
+  [data writeToURL:heartbeatStorageFileURL atomically:YES];
+  XCTAssertNotNil(data);
+
+  // 3. Tag and save new heartbeat date for `storedTag`.
+  NSDate *date = [NSDate date];
+  BOOL successfulSave = [self.storage setHearbeatDate:date forTag:storedTag];
+  XCTAssert(successfulSave);
+
+  // 4. Retrieve saved heartbeat info.
+  NSDate *retrievedDate = [self.storage heartbeatDateForTag:storedTag];
+
+  // 5. Assert that requested heartbeat info matches what was stored.
+  //    This implies the storage directory & file were created, written to, and read from.
+  XCTAssertEqualObjects(retrievedDate, date);
+
+  // 6. Assert that stored heartbeat dictionary has been archived as a mutable dictionary.
+  NSError *error;
+  NSData *objectData = [NSData dataWithContentsOfURL:heartbeatStorageFileURL
+                                             options:0
+                                               error:&error];
+  XCTAssertNotNil(objectData);
+  XCTAssert(objectData.length > 0);
+  XCTAssertNil(error);
+
+  NSSet<Class> *objectClasses = [NSSet setWithArray:@[ NSDictionary.class, NSDate.class ]];
+  NSDictionary *unarchivedDictionary = [GULSecureCoding unarchivedObjectOfClasses:objectClasses
+                                                                         fromData:objectData
+                                                                            error:&error];
+  XCTAssertNotNil(unarchivedDictionary);
+  XCTAssertNil(error);
+  // Unarchived as a mutable dictionary.
+  XCTAssertTrue([unarchivedDictionary isKindOfClass:[NSMutableDictionary class]]);
+}
+
 - (void)testConformsToHeartbeatStorableProtocol {
   XCTAssertTrue([self.storage conformsToProtocol:@protocol(GULHeartbeatDateStorable)]);
 }
 
+#pragma mark - GoogleUtilities Issue #36
+
+- (void)testFix_pre7dot4_CompabitibilityIssue {
```

Comment:
Add this test to validate current fix. This test passes. If the implementation is modified with a one line edit to write an *immutable* dictionary to disk, then this test fails with error from #36